### PR TITLE
IE-339/activate-user-send-email + add --send-email flag

### PR DESCRIPTION
**Pull request by vuthikxkol**
Added --send-email flag to activate-user command which will send an activation email to user which will prompt the user to reset their password.

### CODE REVIEW

**Comment by vuthikxkol** [comment]
@dspasojevic @dbalmain Ready for review

**Comment by dspasojevic** [comment]
@vuthikxkol please revert all the noise in this PR. Keep PRs small. Don't mix style changes with substance changes.

**Comment by vuthikxkol** [comment]
@dspasojevic Its been reverted now

File: src/scripts/services/user-service.ts [hunk 1]
```diff
@@ -196,14 +196,17 @@ export class OktaUserService {
         )}].`
     );
 
-  readonly activateUser = (userId: string): TE.TaskEither<string, User> =>
+  readonly activateUser = (
```
Comment (dspasojevic) [inline src/scripts/services/user-service.ts:199]: **Issue:** document this method please.

File: src/scripts/activate-user.ts [hunk 2]
```diff
@@ -106,30 +106,77 @@ const dryRunActivateUser = (
...
+      )
+    );
+
+/**
+ * Activates the user and sends the activation email.
+ * @param service - the service to use to activate the user.
+ * @param user - the user to activate.
+ * @returns a TaskEither that resolves to the activated user.
+ */
+export const activateUserAndSendEmail =
```
Comment (dspasojevic) [inline src/scripts/activate-user.ts:158]: **Issue:** I don't like boolean flags, but I think having one function that takes a `sendEmail` parameter is better than this duplication.

If your function looks like this:
```
...
someFlag ? 
   doOneThing :
   doAnotherThing
```

it is a code smell and to be considered.

**Comment by dspasojevic** [comment]
> @dspasojevic Its been reverted now

Thanks @vuthikxkol. It is a good habit to look at the PR before adding reviewers. This lets you see the PR in a fresh context.

File: src/scripts/services/user-service.ts (hunk 1 above)
Comment (vuthikxkol) [reply src/scripts/services/user-service.ts:205]: added in commit: 6b672568455dca1ecc7b7d13d462bde70475aa8b

File: src/scripts/activate-user.ts (hunk 2 above)
Comment (vuthikxkol) [reply src/scripts/activate-user.ts]: modified in commit: 6b672568455dca1ecc7b7d13d462bde70475aa8b

**Comment by vuthikxkol** [comment]
Ready for re-review @dspasojevic @dbalmain

File: src/scripts/services/user-service.ts (hunk 1 above)
Comment (dspasojevic) [reply src/scripts/services/user-service.ts:205]: This method needs a description. When you are next in the office, grab the green elements of java style book to read its section on comments.

File: src/scripts/activate-user.ts [hunk 3]
```diff
@@ -187,11 +205,22 @@ export default (
...
       readonly userId: string;
       readonly dryRun: boolean;
+      readonly sendEmail: boolean;
     }) => {
       const client = oktaManageClient({ ...args });
       const service = new OktaUserService(client);
-      const { userId, dryRun } = args;
-      const result = await activateUserHandler(service, userId, dryRun)();
+      const { userId, dryRun, sendEmail } = args;
+      const commandHandler: (
```
Comment (dspasojevic) [inline src/scripts/activate-user.ts:213]: Nice.

**Comment by dspasojevic** [comment]
Looks close @vuthikxkol .

**Comment by vuthikxkol** [comment]
Ready for re-review

**Review by dspasojevic — APPROVED** [review]

**Pull request merged**